### ISSUE TITLE
Improve loading state for table + tree view

### ISSUE THREAD

**Issue by schickling** [prisma/studio#50]
https://www.loom.com/share/9d35fbd9ea3b45b9a7521d9faae3407b

**Comment by kuldar** [comment]
Should this be alleviated with triggering the requets for new rows sooner?

Here's few mockups for "placeholder rows" if we should need them though.

![table-row-placeholders](https://user-images.githubusercontent.com/1710629/63732945-cc40b680-c876-11e9-8d70-0f4431740289.png)
![tree-row-placeholders](https://user-images.githubusercontent.com/1710629/63732946-ccd94d00-c876-11e9-8750-e4684a4e1314.png)

**Comment by madebysid** [comment]
Will add in skeleton rows too, but with a few perf improvements in alpha.160, white spaces like the one you saw should become pretty rare!

**Comment by madebysid** [comment]
IMO let's also defer this until after we have a way to know exactly how many records exist in a table.  I can then show the correct number of skeleton rows. Thoughts?

**Comment by schickling** [comment]
> IMO let's also defer this until after we have a way to know exactly how many records exist in a table. I can then show the correct number of skeleton rows. Thoughts?

I think there's no harm in filling the screen with skeleton rows even if there are less than displayed while loading. Agree @kuldar?

**Comment by schickling** [comment]
> Will add in skeleton rows too, but with a few perf improvements in alpha.160, white spaces like the one you saw should become pretty rare!

Would be great to implement this soon as it's still very common for the cases I've tried. (I guess related to the fix for #106.)

**Comment by idan** [comment]
@schickling @madebysid @kuldar is this being worked on? If not, any notion re: priority?

**Comment by idan** [comment]
tl;dr not sure if to schedule for beta or post-beta

**Comment by madebysid** [comment]
@idan It isn't, but since yesterday's discussion, I'm planning on picking this (and empty states) up soon. IMO pre-beta, since it isn't a whole lot of work anyway.

**Comment by matthewmueller** [comment]
We'll do this for the table view. 

We'll do it for the tree view unless it takes 1 hour or less, otherwise we'll tackle that post-Beta.

**Comment by igalklebanov** [comment]
Hey :wave:

Prisma ORM 7 introduces a brand new standalone [Studio](https://www.prisma.io/docs/orm/tools/prisma-studio), built from the ground up to work everywhere, with or without Prisma ORM. Consequently, all "old Studio" issues are being closed as there won't be any further work on it. :mega:

The new Studio is also available in the Prisma VS Code extension and in the Prisma Console application. :electric_plug:

Help shape, improve and fix the new Studio with us in this repository by submitting bug reports and feature requests. :construction_worker:
